lib/template.html

```diff
@@ -206,28 +209,57 @@
     "headline": {{ seo_page_title | jsonify }},
 {% endif %}
 
+{% if seo_author_name %}
+    "author": {{ seo_author_name | jsonify }},
+{% endif %}
+
 {% if seo_page_image %}
-    "image": {{ seo_page_image | jsonify }},
+    {% if page.image.height && page.image.width %}
+      "image": {
+          "@type": "ImageObject",
+          "url": {{ seo_page_image | jsonify }},
+          "height": {{ page.image.height | jsonify }},
+          "width": {{ page.image.width | jsonify }}
+      },
+    {% else %}
+        "image": {{ seo_page_image | jsonify }},
+    {% endif %}
 {% endif %}
 
 {% if page.date %}
     "datePublished": {{ page.date | date_to_xmlschema | jsonify }},
 {% endif %}
 
+{% if page.dateModified %}
+    "dateModified": {{ page.dateModified | date_to_xmlschema | jsonify }},
+{% elseif page.date %}
+    "dateModified": {{ page.date | date_to_xmlschema | jsonify }},
+{% endif %}
+
 {% if seo_description %}
     "description": {{ seo_description | jsonify }},
 {% endif %}
 
 {% if seo_site_logo %}
     "publisher": {
         "@type": "Organization",
+        {% if seo_author_name %}
+            "name": {{ seo_author_name | jsonify }},
+        {% endif %}
         "logo": {
             "@type": "ImageObject",
             "url": {{ seo_site_logo | jsonify }}
         }
     },
 {% endif %}
 
+{% if seo_type == "BlogPosting" %}
+    "mainEntityOfPage": {
+         "@type": "WebPage",
+         "@id": {{ page.url | replace:'/index.html','/' | absolute_url | jsonify }}
+    },
+{% endif %}
+
```

Comment:
Add mainEntityOfPage when BlogPosting type is chosen. 
Better explained in this [StackOverflow post](http://stackoverflow.com/a/34467088/1813592).